### ISSUE TITLE
ITTNotify usage must be configurable

### ISSUE THREAD

**Issue by ilya-lavrenov** [oneapi-src/oneTBB#654]
When static library build is used, ITT notify symbols may conflict with other ITT notify symbols from end application / target library.
TBB must provide a way to switch ITT off

**Comment by alexey-katranov** [comment]
oneTBB does not officially support static build due to potential issues with symbol conflicts, correctness and performance. However, you can try to disable `__TBB_USE_ITT_NOTIFY` macro ([src/tbb/CMakeLists.txt#L65](https://github.com/oneapi-src/oneTBB/blob/master/src/tbb/CMakeLists.txt#L65)) to disable ITT instrumentation (pay attention that the instrumentation is used by profiling tools for better analysis.)

**Comment by ilya-lavrenov** [comment]
Yes, I did this, but it require TBB modifications while it would be nice to have an explicit cmake option

**Comment by alexey-katranov** [comment]
> Yes, I did this, but it require TBB modifications while it would be nice to have an explicit cmake option

The problem with any additional option is that it significantly increases the number of configurations to test (this is kind of a combinatorial problem). If you are interested, we can come up with some untested option, but from time to time it can break.

> In production build where all performance issues are investigated / fixed - it's not needed.

ITT instrumentation is invented for production builds that our users can have additional information when profiling their applications. So, you do not need a special build to profile your oneTBB based application.

**Comment by isaevil** [comment]
@ilya-lavrenov is it fine for you if make `-DTBB_USE_ITT_NOTIFY` option for CMake but leave it untested?

**Comment by ilya-lavrenov** [comment]
> @ilya-lavrenov is it fine for you if make `-DTBB_USE_ITT_NOTIFY` option for CMake but leave it untested?

Yes, sure

**Comment by ilya-lavrenov** [comment]
What is current status of this? is WIP or not?

**Comment by arunparkugan** [comment]
@ilya-lavrenov is this issue still relevant?

**Comment by ilya-lavrenov** [comment]
Yes
